src/java/org/apache/cassandra/tools/nodetool/ClientStats.java

```diff
@@ -86,54 +92,51 @@ public void execute(NodeProbe probe)
             return;
         }
 
-        if (listConnections)
+        // Note: for compatbility with existing implementation if someone passes --all (listConnections),
+        // --client-options, and --metadata all three will be printed.
+        List<Map<String, String>> clients = (List<Map<String, String>>) probe.getClientMetric("connections");
+        if (!clients.isEmpty() && (listConnections || clientOptions || verbose))
         {
-            List<Map<String, String>> clients = (List<Map<String, String>>) probe.getClientMetric("connections");
-            if (!clients.isEmpty())
+            List<String> tableHeaderBase = Lists.newArrayList("Address", "SSL", "Cipher", "Protocol", "Version",
```

Comment:
NIT: prefer JDK APIs. Since trunk and 5.0 are on Java 11+ 
```suggestion
            List<String> tableHeaderBase = List.of("Address", "SSL", "Cipher", "Protocol", "Version",
```